### PR TITLE
guilt: init at 0.37-rc1

### PR DESCRIPTION
**Pull request by JaviMerino**
## Description of changes

[guilt](https://github.com/jeffpc/guilt) lets you manage patches like quilt, on top of a git repository. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - Yes
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/guilt/default.nix [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+, lib
+, makeWrapper
+, perl
+, stdenv
+, xmlto
+}:
+
+let
+  version = "0.37-rc1";
+  sha256 = "sha256-7OgRbMGYWtGvrZxKfJe0CkpmU3AUkPebF5NyTsfXeGA=";
```
Comment (matthiasbeyer) [inline pkgs/applications/version-management/guilt/default.nix:17]: Please put these into the expression. There's no need here to put these in a `let`!

File: pkgs/applications/version-management/guilt/default.nix [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "jeffpc";
+    repo = "guilt";
+    rev = "v${version}";
+    inherit sha256;
+  };
+
+  doCheck = true;
```
Comment (matthiasbeyer) [inline pkgs/applications/version-management/guilt/default.nix:30]: Is this required? Isn't it true by default?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]

File: pkgs/applications/version-management/guilt/default.nix (hunk 2 above)
Comment (JaviMerino) [reply pkgs/applications/version-management/guilt/default.nix:30]: [It is not](https://nixos.org/manual/nixpkgs/stable/#var-stdenv-doCheck).

File: pkgs/applications/version-management/guilt/default.nix (hunk 1 above)
Comment (JaviMerino) [reply pkgs/applications/version-management/guilt/default.nix]: Done!

File: pkgs/applications/version-management/guilt/default.nix (hunk 2 above)
Comment (matthiasbeyer) [reply pkgs/applications/version-management/guilt/default.nix:26]: Ah. TIL! Thanks!

**Comment by matthiasbeyer** [comment]
@ofborg build guilt

**Comment by JaviMerino** [comment]
@matthiasbeyer I think I have addressed all the comments. Can you have another look?

**Review by matthiasbeyer — APPROVED** [review]

**Comment by JaviMerino** [comment]
@matthiasbeyer What else do I have to do to get this merged?

**Pull request merged**